### PR TITLE
Add Schnorr Protocol exercise

### PR DESCRIPTION
**Pull request by surfer05**

### CODE REVIEW

File: exercises/zk-protocols-basics/02_schnorr_protocol/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+Before diving into the protocol, it's crucial to understand some mathematical concepts:
+
+- **Modular Arithmetic:** Arithmetic operations performed with respect to a modulus $n$, where numbers "wrap around" upon reaching $n$.
+- **Finite Fields:** A set of numbers with addition, subtraction, multiplication, and division operations defined, satisfying certain properties.
+- **Discrete Logarithm Problem:** Given $g$ and $y$, find $x$ such that $g^x=y\ mod p$. This problem is computationally hard for large primes, which provides security in cryptographic protocols.
+
+**Key Parameters:**
+
+- $p$: A large prime number.
+- $g$: A generator of the multiplicative group, meaning $g$ is a primitive root modulo $p$.
```
Comment (rutefig) [inline exercises/zk-protocols-basics/02_schnorr_protocol/README.md:20]: Maybe we should also introduce briefly what is a generator on the **Mathematical Background**

File: exercises/zk-protocols-basics/02_schnorr_protocol/README.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+   - Verifier computes:
+     - $\text{left} = g^s \mod p$.
+     - $\text{right} = t \cdot y^c \mod p$.
+   - Verifier checks if $\text{left} \equiv \text{right} \mod p$.
+     - If equal, the proof is accepted.
+     - If not, the proof is rejected.
+
+---
+
+## Detailed Example
```
Comment (rutefig) [inline exercises/zk-protocols-basics/02_schnorr_protocol/README.md:61]: This is pretty cool 🚀

**Review by rutefig — CHANGES_REQUESTED** [review]
Just small improvements to be done

Good job 👏🏼

File: exercises/zk-protocols-basics/02_schnorr_protocol/README.md (hunk 1 above)
Comment (surfer05) [reply exercises/zk-protocols-basics/02_schnorr_protocol/README.md:21]: Made the required changes, please review.

**Review by rutefig — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**